### PR TITLE
feat(design): added custom injectors to DaffModal

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: https://github.com/graycoreio/daffodil/issues/938

## Does this PR introduce a breaking change?
```
[x]Yes
[ ] No
```

Currents breaks the Modal implementation in demo and any other potential use.
## Other information

### CODE REVIEW

**Comment by Nolan-Arnold** [comment]
For issue #938

File: libs/design/src/molecules/modal/modal.ts [hunk 1]
```diff
@@ -2,7 +2,12 @@ import { ComponentRef } from '@angular/core';
 import { DaffModalComponent } from './modal/modal.component';
 import { OverlayRef } from '@angular/cdk/overlay';
 
-export interface DaffModal {
+export class DaffModal<T> {
+
```
Comment (damienwebdev) [inline libs/design/src/molecules/modal/modal.ts:6]: This function exists on the modal service. This is really more of a helper function. I don't think we should be providing two ways of doing the same thing.

Comment (Nolan-Arnold) [reply libs/design/src/molecules/modal/modal.ts:6]: changed in 93fdd37ad40ae5d86f9f21688d883ca8854b93aa. We could potentially add a directive to go on a button, similar to material, that would close the modal.

File: libs/design/src/molecules/modal/service/modal.service.ts [hunk 2]
```diff
@@ -59,22 +60,22 @@ export class DaffModalService {
...
-		const _attachedModal = this._attachModalContent(component, _modal);
+		const modal = new DaffModal<T>(_modal, _ref);
+		const injector = this.createInjector<T>(modal);
 
-		const modal: DaffModal = {
-			modal: _modal,
-			overlay: _ref,
-		};
+
+		const _attachedModal = this._attachModalContent(component, _modal, injector);
```
Comment (damienwebdev) [inline libs/design/src/molecules/modal/service/modal.service.ts:74]: This behavior needs a test.

**Comment by damienwebdev** [comment]
Closing as stale.

**Pull request closed without merging**